R2API.Skins/SkinVFX.cs

```diff
@@ -0,0 +1,229 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Mono.Cecil.Cil;
+using MonoMod.Cil;
+using RoR2;
+using UnityEngine;
+using RoR2.Orbs;
+
+namespace R2API;
+
+/// <summary>
+/// Class for adding skin-specific effect replacements for SkinDefs.
+/// </summary>
+public static partial class SkinVFX {
+    private static List<SkinVFXInfo> skinVFXInfos = new List<SkinVFXInfo>();
+    private static bool hooksSet = false;
+    private const uint BaseIdentifier = 24000; // arbitrary, but we shouldn't hit 24,000 unique items for a substantial amount of time.
+    private static uint currentIdentifier = BaseIdentifier; 
+    private static uint nextIdentifier => currentIdentifier++;
```

Comment:
Could use a bitflag instead of tracking id numbers,should make things simpler all around.

Reply:
tad confused, what exactly would a bitflag do here? the identifier is only ever one value, it never needs to be compared with OR/AND, and the first ~10-12 bits would need to be reserved for PickupIndexes (as scrappers and item steal orbs utilize the genericUint field to store them)

Reply:
It'd allow you to claim only one bit of the uint,instead of any value +24000,you'd lose the FindSkinVFX by id overload but afaict that isn't used anywhere that the other one couldn't and doesn't seem to provide any perf benefit,since it's still a search instead of a direct lookup.
To be honest claiming any part of the effectdata genericUint feels slightly icky since we can't actually know what it's being used for before modifying it (even though vanilla only uses it for ItemIndexes,for now (who knows if the dlc will use it (it probably won't))) but I understand why it's beneficial.
Though if we _are_ working off the assumption that the effectdata values won't be used outside of our knowledge the genericBool might be an even better choice,since it's not used by vanilla at all :thinking:

Reply:
the overload with identifier is used in ApplyModifier and ApplyReplacement, to pass along "this matched a skin when fired, and should be replaced with the appropriate effect", since EffectComponent stores no information on who fired it

cant think of how i'd do that with a bitflag without being limited to just 20 effect replacements (since the first 12 or so bits can't be used, as they might be a scrapper pickup orb instead)

Reply:
Fair enough,for some reason I was thinking that those hooks had access to the gameobject that was creating the effect.If that was the case,one bit of info would have been enough.